src/error.h

```diff
@@ -145,15 +154,49 @@ ErrorReturnVoid(const Char * msg, Int arg1, Int arg2, const Char * msg2);
 *F  RequireArgumentConditionMayReplace
 */
 #define RequireArgumentConditionMayReplace(funcname, op, argname, cond, msg) \
-    do { \
-        while (!(cond)) { \
-            op = ErrorReturnObj(funcname ": <" argname "> " msg " (not a %s)", \
-                (Int)TNAM_OBJ(op), 0, \
-                "you can replace <" argname "> via 'return <" argname ">;'"); \
-        } \
+    do {                                                                     \
+        while (!(cond)) {                                                    \
+            op = RACErrorHelper(1, funcname, op, argname, msg);              \
+        }                                                                    \
     } while (0)
 
 
+/****************************************************************************
+**
+*F  RequireInt
+*/
+#define RequireInt(funcname, op, argname) \
+    RequireArgumentCondition(funcname, op, argname, IS_INT(op), \
+        "must be an integer")
+
+
+/****************************************************************************
+**
+*F  RequireIntMayReplace
+*/
+#define RequireIntMayReplace(funcname, op, argname) \
+    RequireArgumentConditionMayReplace(funcname, op, argname, IS_INT(op), \
+        "must be an integer")
+
+
+/****************************************************************************
+**
+*F  RequireSmallInt
+*/
+#define RequireSmallInt(funcname, op, argname) \
+    RequireArgumentCondition(funcname, op, argname, IS_INTOBJ(op), \
+        "must be an integer")
```

Comment:
Should the refer to the "smallness" of the required integer somehow?

Reply:
Ouch, indeed, that's a copy&past mistake. I'll change it to "must be a small integer"